clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -933,10 +933,15 @@ def BinOp : CIR_Op<"binop", [Pure,
   // TODO: get more accurate than CIR_AnyType
   let results = (outs CIR_AnyType:$result);
   let arguments = (ins Arg<BinOpKind, "binop kind">:$kind,
-                       CIR_AnyType:$lhs, CIR_AnyType:$rhs);
+                       CIR_AnyType:$lhs, CIR_AnyType:$rhs,
+                       UnitAttr:$hasNoUnsignedWrap,
+                       UnitAttr:$hasNoSignedWrap);
 
   let assemblyFormat = [{
-    `(` $kind `,` $lhs `,` $rhs  `)` `:` type($lhs) attr-dict
+    `(` $kind `,` $lhs `,` $rhs  `)` 
+    (`nsw` $hasNoSignedWrap^)?
```

Comment:
Add a verifier to make sure both `nsw` and `nuw` are only valid for the binops that make sense. It might also be needed to check for valid types: are those allowed with vectors or (recently introduced) extInts?

Reply:
added!

Reply:
This allowed with any integer type - and wtth extInts as well (checked). But is no allowed with vectors